ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/activeMapping/ContinuousPlanningTools.java

```diff
@@ -18,6 +21,22 @@
 
 public class ContinuousPlanningTools
 {
+   public static double getDistanceFromRobotToGoalPoseOnXYPlane(Vector3DBasics robotPositionInWorld, SideDependentList<FramePose3D> goalPoses)
+   {
+      FramePose3D leftGoalPose = goalPoses.get(RobotSide.LEFT);
+      FramePose3D rightGoalPose = goalPoses.get(RobotSide.RIGHT);
+
+      // Get point halfway between the left and right goal poses
+      leftGoalPose.interpolate(rightGoalPose, 0.5);
+      FramePose3D middleDistanceBetweenGoalPoses = leftGoalPose;
+      middleDistanceBetweenGoalPoses.getPosition().setZ(0.0);  // We want to check distance in XY Plane so zero the Z
```

Comment:
these few lines are actually pretty problematic, I think. You're changing leftGoalPose Pose for no real reason, which I'm guessing is going to mess up some of the extgernal stuff.

I would change to:

```
Point3D middleDistanceBetweenGoalPoses = new Point3D();
middleDistanceBetweenGoalPoses.interpolate(leftGoalPose.getPosition(), rightGoalPose.getPosition(), 0.5);
```

Reply:
then setting the height to zero is irrelevant if you use `distanceXY` instead of `distance`

Reply:
True thats very good